File: 백준/Silver/1235. 학생 번호/학생 번호.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        String[] students = new String[n];

        for(int i = 0;i<n;i++){
            students[i] = br.readLine().trim();
        }
        int end = 1;
        while (true) {
            Set<String> res = new HashSet<>();
            for(String student: students){
                int len = student.length();
                res.add(student.substring(Math.max(0,len-end)));
            }
            if(res.size()==n){
                System.out.println(end);
                break;
            }
            end++;
            
        }
    }
}

File: 백준/Silver/1235. 학생 번호/학생 번호.py
import sys
input = sys.stdin.readline

n = int(input())
students = [input().strip() for _ in range(n)]

end = 1
while True:
    res = set()
    for student in students:
        res.add(student[-end:])#끝에서부터 end길이만큼
    if len(res) ==n:
        print(end)
        break

    end+=1


